fix(build): prevent adding `only-in-en-us` css class to external links

yin1999 opened the pull request:
## Summary

Fixes: mdn/rari#348

### Problem

The resolved redirection link may also be an external one. But we haven't check this before assigning the `enUSFallbackURL`, which caused the `only-in-en-us` CSS class to be added accidentally to the external links.

### Solution

Check whether the resolved URL is an external one, if so, do not assign the `enUSFallbackURL`. Which means, the `only-in-en-us` CSS class would not be added.

---

## Screenshots

Checkout the git commit sha `f277be087b9484b077fc734e5192d83bc3ee04a1`, which is the last commit sha before the fix in translated-content (mdn/translated-content#21711).

Check the section `/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Your_first_WebExtension#打包和发布`:

### Before

See: mdn/rari#348

### After

![image](https://github.com/mdn/yari/assets/15844309/e09688ea-8c7d-4b3a-ac27-63e86903ce73)

---

## How did you test this change?

Run `yarn dev`, and check the rendered page.

yin1999 committed:
fix(build): prevent adding `only-in-en-us` css class to external links

yin1999 committed:
add a test case

argl commented:
This seems to be resolved with switch to rari rendering.

<img width="824" alt="image" src="https://github.com/user-attachments/assets/88ba199c-3dea-4a6e-b841-79e956c42be7">

Closing this.

yin1999 commented:
> This seems to be resolved with switch to rari rendering.

Hey, @argl. I'm not sure if I tested it correctly, but when I updated `@mdn/rari` in mdn/content to version `v0.1.0` and then ran `yarn start:rari`, I could see that the link was broken (when I replaced this link with the one we should have a redirect for it before, that is, `/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Publishing_your_WebExtension`).

![image](https://github.com/user-attachments/assets/bcd4aa92-94b3-4843-94c1-03a9e8c154fd)
